### PR TITLE
Replace trace! with debug! calls in machine module

### PR DESCRIPTION
**Pull request by loony-bean**
Those logs are more important than just traces

### CODE REVIEW

**Review by zeenix — CHANGES_REQUESTED** [review]
I do agree with some of the changes but not most. As I [pointed out in the closed PR](https://github.com/blockjoy/blockvisor/pull/148#pullrequestreview-1090276040), the solution to avoid noise from irrelevant sources is to control the logging level more specifically. Upgrading our own tracing output to debug output, isn't the best way IMO.

**Comment by loony-bean** [comment]
> I do agree with some of the changes but not most. As I [pointed out in the closed PR](https://github.com/blockjoy/blockvisor/pull/148#pullrequestreview-1090276040), the solution to avoid noise from irrelevant sources is to control the logging level more specifically. Upgrading our own tracing output to debug output, isn't the best way IMO.

Which `trace` calls do you think could be changed to `debug`?

**Comment by zeenix** [comment]
> Which `trace` calls do you think could be changed to `debug`?

I'll comment with `+` and `-` to indicate.

File: src/machine.rs [hunk 1]
```diff
@@ -49,7 +49,7 @@ impl<'m> Machine<'m> {
     pub async fn create(config: Config<'m>) -> Result<Machine<'m>, Error> {
         let vm_id = *config.vm_id();
         info!("Creating new machine with VM ID `{vm_id}`");
-        trace!("{vm_id}: Configuration: {:?}", config);
+        debug!("{vm_id}: Configuration: {:?}", config);
```
Comment (zeenix) [inline src/machine.rs:52]: +

File: src/machine.rs [hunk 2]
```diff
@@ -62,7 +62,7 @@ impl<'m> Machine<'m> {
             .await?;
 
         let dest = config.kernel_image_path();
-        trace!(
+        debug!(
```
Comment (zeenix) [inline src/machine.rs:65]: :+1:

File: src/machine.rs [hunk 3]
```diff
@@ -211,7 +211,7 @@ impl<'m> Machine<'m> {
             .stdin(stdin)
             .stdout(stdout)
             .stderr(stderr);
-        trace!("{vm_id}: Running command: {:?}", cmd);
+        debug!("{vm_id}: Running command: {:?}", cmd);
```
Comment (zeenix) [inline src/machine.rs:214]: :-1:

File: src/machine.rs [hunk 4]
```diff
@@ -230,27 +230,27 @@ impl<'m> Machine<'m> {
         if let Err(e) = self
             .setup_vm()
             .and_then(|_| async {
-                trace!("{vm_id}: Booting the VM instance...");
+                debug!("{vm_id}: Booting the VM instance...");
```
Comment (zeenix) [inline src/machine.rs:233]: :-1:

File: src/machine.rs [hunk 5]
```diff
@@ -230,27 +230,27 @@ impl<'m> Machine<'m> {
...
+            debug!(
                 "{vm_id}: Failed to boot VM instance: {}. Force shutting down..",
                 e
             );
             self.force_shutdown().await.unwrap_or_else(|e| {
                 // `force_shutdown` only updates the state on success.
                 self.state = MachineState::SHUTOFF;
                 // We want to return to original error so only log the error from shutdown.
-                trace!("{vm_id}: Failed to force shutdown: {}", e);
+                debug!("{vm_id}: Failed to force shutdown: {}", e);
```
Comment (zeenix) [inline src/machine.rs:247]: Actually I think this and the previous one should be a warning instead.

File: src/machine.rs [hunk 6]
```diff
@@ -230,27 +230,27 @@ impl<'m> Machine<'m> {
...
             });
 
             return Err(e);
         }
 
-        trace!("{vm_id}: VM started successfully.");
+        debug!("{vm_id}: VM started successfully.");
```
Comment (zeenix) [inline src/machine.rs:253]: :-1:  here too. The caller knows about the success so they can log the success on any level they prefer.

File: src/machine.rs [hunk 7]
```diff
@@ -296,7 +296,7 @@ impl<'m> Machine<'m> {
                 // In case of tmux, we need to kill the tmux session.
                 let cmd = &mut Command::new("tmux");
                 cmd.args(&["kill-session", "-t", &session_name]);
-                trace!("{vm_id}: Running command: {:?}", cmd);
+                debug!("{vm_id}: Running command: {:?}", cmd);
```
Comment (zeenix) [inline src/machine.rs:299]: :-1:

File: src/machine.rs [hunk 8]
```diff
@@ -261,7 +261,7 @@ impl<'m> Machine<'m> {
     #[instrument(skip_all)]
     pub async fn force_shutdown(&mut self) -> Result<(), Error> {
         let vm_id = self.config.vm_id();
-        trace!("{vm_id}: Killing VM...");
+        debug!("{vm_id}: Killing VM...");
```
Comment (zeenix) [inline src/machine.rs:264]: :-1:

File: src/machine.rs [hunk 9]
```diff
@@ -329,18 +329,18 @@ impl<'m> Machine<'m> {
 
         if let MachineState::RUNNING { .. } = self.state {
             if let Err(err) = self.shutdown().await {
-                trace!("{vm_id}: Shutdown error: {err}");
+                debug!("{vm_id}: Shutdown error: {err}");
```
Comment (zeenix) [inline src/machine.rs:332]: warning better here too I think.

File: src/machine.rs [hunk 10]
```diff
@@ -287,7 +287,7 @@ impl<'m> Machine<'m> {
                 if !killed {
                     return Err(Error::ProcessNotKilled(pid));
                 }
-                trace!("{vm_id}: Successfully sent KILL signal to VM (pid: `{pid}`).");
+                debug!("{vm_id}: Successfully sent KILL signal to VM (pid: `{pid}`).");
```
Comment (zeenix) [inline src/machine.rs:290]: :-1:

**Comment by zeenix** [comment]
It takes time to decide for each case so I'll stop now and hope my comments will help you decided. The main point is that we shouldn't just upgrade all trace to debug w/o a thought.

**Comment by loony-bean** [comment]
Closing as outdated

**Pull request closed without merging**